### PR TITLE
wait for test and view ???

### PR DESCRIPTION
**Pull request by seafruit**

### CODE REVIEW

**Comment by goddessting** [comment]
:+1:

**Comment by xudaye** [comment]
:+1:

File: app/api/users.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          if (err) return next(err);
+          res.send(201);
+        });
+      }
+    })
+  } else {
+    res.send(400);
+  }
+});
+// exports.register = function (req, res) {
```
Comment (baixue1016) [inline app/api/users.js]: 注释删掉

File: app/api/users.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+//       }
+//     })
+//   } else {
+//     res.send(400);
+//   }
+// };
+
+function validate(user) {
+  if (user.username.length === 8 && (user.password.length <= 10 && user.password.length >= 6)) {
+    console.log(/^[0-9]*$/.test(user.username)+'         0000000000000000000000000000000');
```
Comment (baixue1016) [inline app/api/users.js]: 调试代码删掉

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Delicious School
 =================
 
-CI status: ![Build Status](https://travis-ci.org/js-demos/es6-project-demo.png?branch=master)
+CI status: ![Build Status](https://travis-ci.org/seafruit/delicious-school-project.svg?branch=the-first-week-mvp)
```
Comment (freewind) [inline README.md:4]: master ???

File: app/api/api.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Created by yoyo on 16-8-18.
+ */
```
Comment (freewind) [inline app/api/api.js:3]: delete

File: app/api/users.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Created by yoyo on 16-8-18.
+ */
```
Comment (freewind) [inline app/api/users.js:3]: delete

File: app/api/users.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import express from 'express';
+const router = express.Router();
+let User = require('../../entity/user');
```
Comment (freewind) [inline app/api/users.js:7]: /app

Comment (freewind) [inline app/api/users.js:7]: const

File: app/api/users.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+router.post('/',function(req,res,next){
+  const requestUser = {
+    username: req.body.username,
+    password: req.body.password
+  };
+  let isValidate = validate(requestUser);
```
Comment (freewind) [inline app/api/users.js:14]: const

File: app/api/users.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  if (isValidate) {
+    User.findOne(requestUser, function (err, users) {
```
Comment (freewind) [inline app/api/users.js:16]: 只使用username

Comment (freewind) [inline app/api/users.js:16]: user

File: app/api/users.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Created by yoyo on 16-8-18.
+ */
+
+import express from 'express';
+const router = express.Router();
+let User = require('../../entity/user');
+
+router.post('/',function(req,res,next){
```
Comment (freewind) [inline app/api/users.js:9]: 格式化。。。。

File: app/api/users.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  const requestUser = {
+    username: req.body.username,
+    password: req.body.password
+  };
+  let isValidate = validate(requestUser);
+  if (isValidate) {
+    User.findOne(requestUser, function (err, users) {
+      if (err) return next(err);
+      if (users) {
+        res.send(409);
```
Comment (freewind) [inline app/api/users.js:19]: return res.sendStatus(409)

File: app/api/users.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+      } else {
+        const user = new User({username: requestUser.username, password: requestUser.password});
```
Comment (freewind) [inline app/api/users.js:21]: new User(requestUser)

File: app/api/users.js [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        user.save(function (err) {
+          if (err) return next(err);
+          res.send(201);
```
Comment (freewind) [inline app/api/users.js:24]: sendStatus

File: app/api/users.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        });
+      }
+    })
+  } else {
+    res.send(400);
```
Comment (freewind) [inline app/api/users.js:29]: sendStatus

File: app/api/users.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+});
+
+function validate(user) {
+  if (user.username.length === 8 && (user.password.length <= 10 && user.password.length >= 6)) {
+    if (/^[0-9]*$/.test(user.username)) {
+      return true;
+    }
+  }
+  return false;
+}
```
Comment (freewind) [inline app/api/users.js:40]: /shared/validation.js

export {
  validateUsername,
  validatePassword
}

File: app/api/api.js [hunk 15]
```diff
@@ -0,0 +1,10 @@
...
+import express from 'express';
+import usersApi from './users';
+
+const router = express.Router();
+
+router.use('/users', usersApi);
+export default router;
```
Comment (freewind) [inline app/api/api.js:10]: /app/api/index.js

File: app/server.js [hunk 16]
```diff
@@ -27,13 +33,16 @@ app.use(webpackHotMiddleware(compiler, {
...
 app.post('/login', execute.findUser);
-app.post('/register', execute.register);
+// app.post('/api/users', users.register);
 app.post('/init', execute.findDish);
 app.post('/mealInfo', execute.finsDishInfoById);
 app.post('/saveOrder', execute.saveOrder);
 
+app.use('/api',apiRouter);
+
 app.listen(3000, function () {
```
Comment (freewind) [inline app/server.js:43]: db.connect();

File: app/server.js [hunk 17]
```diff
@@ -4,6 +4,12 @@ import webpackDevMiddleware from 'webpack-dev-middleware';
 import webpackHotMiddleware from 'webpack-hot-middleware';
 import express from 'express';
 import execute from '../mongodb/execute';
+// import users from '../mongodb/users';
+import apiRouter from './api/api';
+
+let db = require('../mongodb/connect');
+db.connect();
+
 let bodyParser = require("body-parser");
```
Comment (freewind) [inline app/server.js:13]: import from

File: app/server.js [hunk 18]
```diff
@@ -4,6 +4,12 @@ import webpackDevMiddleware from 'webpack-dev-middleware';
 import webpackHotMiddleware from 'webpack-hot-middleware';
 import express from 'express';
 import execute from '../mongodb/execute';
+// import users from '../mongodb/users';
+import apiRouter from './api/api';
+
+let db = require('../mongodb/connect');
```
Comment (freewind) [inline app/server.js:10]: import from

File: app/server.js [hunk 19]
```diff
@@ -27,13 +33,16 @@ app.use(webpackHotMiddleware(compiler, {
 
 app.use(express.static('./public'));
 app.post('/login', execute.findUser);
```
Comment (freewind) [inline app/server.js:35]: delete?

File: app/server.js [hunk 20]
```diff
@@ -27,13 +33,16 @@ app.use(webpackHotMiddleware(compiler, {
...
-app.post('/register', execute.register);
+// app.post('/api/users', users.register);
 app.post('/init', execute.findDish);
 app.post('/mealInfo', execute.finsDishInfoById);
 app.post('/saveOrder', execute.saveOrder);
```
Comment (freewind) [inline app/server.js:39]: will delete

File: public/register.jsx [hunk 21]
```diff
@@ -1,31 +1,16 @@
...
-        } else {
-          alert("用户名已存在！");
-          location.href = "/#/register";
-        }
-      }
-
-    })
+    this.state = {
+      username: '',
+      password:''
```
Comment (freewind) [inline public/register.jsx:12]: format

File: public/register.jsx [hunk 22]
```diff
@@ -41,24 +26,52 @@ export default class Register extends Component {
...
             </form>
-            <button type="button" onClick={this.commit} className="btn btn-primary btn-block btn-register">注册</button>
           </div>
           <div className="col-md-4"></div>
         </div>
       </div>
     )
   }
+
+  _onUsernameChange(event){
```
Comment (freewind) [inline public/register.jsx:50]: Changed

File: spec/dummy-spec.js [hunk 23]
```diff
@@ -4,4 +4,4 @@ describe('dummy', () => {
   it('1+1 == 2', () => {
     expect(1+1).toEqual(2);
   })
-});
\ No newline at end of file
+});
```
Comment (freewind) [inline spec/dummy-spec.js:7]: delete

**Pull request merged**